lib/CloudinaryPlugin.js

```diff
@@ -177,12 +169,13 @@ class CloudinaryPlugin {
    */
   async rename(request) {
     const from = this._getArg(request, 'from_public_id'), to = this._getArg(request, 'to_public_id');
-    return await Cloudinary.uploader.rename(from, to)
-      .catch(error => {
-        if (error.http_code == 404) {
-          throw new this.context.errors.NotFoundError(error.message);
-        }
-      })
+    try {
+      return await Cloudinary.uploader.rename(from, to);
+    } catch (error) {
+      if (error.http_code === 404) {
```

Comment:
If the error is not a 404, you're discarding the exception and returning an `undefined` value.